examples/barotropic_tide.jl

```diff
@@ -0,0 +1,285 @@
+# # Barotropic tide
+#
+# In this example, we simulate the evolution of a barotropic tide over a hill.
+#
+# ## Install dependencies
+#
+# First let's make sure we have all required packages installed.
+
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, CairoMakie"
+# ```
+
+using Oceananigans
+using Oceananigans.Units
+
+# ## Grid
+
+# We use an immersed boundary two-dimensional grid (in ``x``--``z``) that is is periodic in
+# the ``x``-direction. To construct an immersed boundary grid we first need to create what
+# we refer to as "underlying grid", which the grid that encompasses the immersed boundary.
+
+Nx, Nz = 200, 80
+
+H  = 2kilometers
+Lx = 2000kilometers
+
+underlying_grid = RectilinearGrid(size = (Nx, Nz),
+                                  x = (-Lx/2, Lx/2),
+                                  z = (-H, 0),
+                                  halo = (4, 4),
+                                  topology = (Periodic, Flat, Bounded))
+
+# Now we can create the non-trivial bathymetry as an immersed boundary. We use `GridFittedBottom` that
+# gets as input either *(i)* a two-dimensional function whose arguments are the grid's native horizontal
+# coordinates and it returns the ``z`` of the bottom, or *(ii)* a two-dimensional array with the values
+# of ``z`` at the bottom at the cell centers.
+#
+# In this example we'd like to have a small Gaussian hill at the center of the domain.
+
+# ```math
+# h(x) = -H + h_0 \exp(-x^2 / 2σ^2)
+# ```
+
+h₀ = 50 # m
+width = 5kilometers
+hill(x) = h₀ * exp(-x^2 / 2width^2)
+bottom(x, y) = - H + hill(x)
+
+grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(bottom))
```

Comment:
This example is probably the perfect test for `PartialCellBottom`

Reply:
Yes, or even better for shaved cells!
I'd be very interested to see if these streaks of w-velocities in the initial condition diminish with shaved cells.

Reply:
cc @siddharthabishnu

Reply:
Why don't we use `PartialCellBottom` here?

Reply:
> We can delete this validation test too after this: https://github.com/CliMA/Oceananigans.jl/blob/main/validation/immersed_boundaries/internal_tide.jl

The validation script includes a multi region example. Shall I still delete it?
I could also clean it up and add toml files in the validation script directory (see #3076).

Reply:
When I tried to change

```Julia
grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(bottom))
```

to 

```Julia
using Oceananigans.ImmersedBoundaries: PartialCellBottom
grid = ImmersedBoundaryGrid(underlying_grid, PartialCellBottom(bottom))
```

the simulation `NaN`ed immediately. Even when I reduced the `max_Δt` from 6 minutes down to 2 minutes..